pulp_2to3_migration/app/plugin/rpm/pulp_2to3_models.py

```diff
@@ -189,11 +193,62 @@ class Pulp2Erratum(Pulp2to3Content):
 
     pulp2_type = 'erratum'
     set_pulp2_repo = True
+    cached_repo_data = {}
 
     class Meta:
         unique_together = ('errata_id', 'repo_id')
         default_related_name = 'erratum_detail_model'
 
+
+    @classmethod
+    def get_repo_data(cls, pulp2_repo):
+        """
+        Get content data of a repository, NEVRA of packages and NSVCA of modules.
+
+        Args:
+            pulp2_repo(Pulp2Repository): a pre-migrated repo to collect data from
+
+        Returns:
+            dict: {'packages': list of NEVRA tuples of the packages which are in this repo,
+                   'modules': list of NSVCA tuples of the module which are in this repo}
+
+        """
+        if pulp2_repo.pk in cls.cached_repo_data:
+            return cls.cached_repo_data[pulp2_repo.pk]
+
+        repo_pkg_nevra = []
+        repo_module_nsvca = []
+
+        # gather info about available packages
+        package_pulp2_ids = Pulp2RepoContent.objects.filter(
+            pulp2_repository=pulp2_repo,
+            pulp2_content_type_id='rpm'
+        ).only('pulp2_unit_id').values_list('pulp2_unit_id', flat=True)
+
+        pulp2rpms = Pulp2Rpm.objects.filter(pulp2_id__in=package_pulp2_ids)
+        for pkg in pulp2rpms.iterator():
+            repo_pkg_nevra.append((pkg.name, pkg.epoch or '0', pkg.version, pkg.release, pkg.arch))
+
+        # gather info about available modules
+        modulemd_pulp2_ids = Pulp2RepoContent.objects.filter(
+            pulp2_repository=pulp2_repo,
+            pulp2_content_type_id='modulemd'
+        ).only('pulp2_unit_id').values_list('pulp2_unit_id', flat=True)
```

Comment:
Is there a particular reason why to use values_list() in a couple of places?

Reply:
@dralley , just a way to construct a list of values to feed in the another query.
Any alternative suggestions are welcome.

Reply:
Actually, never mind, I was going to say that you can just pass in the queryset, but I forgot that that only works with the PK, wheras here we actually have "pulp2_unit_id", so this is in fact probably the simplest way to do it.